### DISCUSSION TITLE
Windows needs a cleaner git installation

### DISCUSSION THREAD

**Discussion by just1a-person** [microsoft/Windows-Dev-Performance#72] [Ideas]
Currently installing git will install things like Git Bash, Git GUI, MinTTY and a lot of recompiled GNU utilities for Windows and other ~*bloat*~ *unnecessary things*. I just want git the executable in PATH.

Of course, I can choose what I want to install in the installer but when I use a tool like `winget`, this doesn't really make sense.

I don't know if Microsoft can do anything about it. But I am just mentioning it here because the README says:

> The purpose of this repo is to collect and catalog issues that affect / impact developers who work on Windows

If anyone wants these tools like Bash (compiled for Windows), MinTTY, recompiled GNU utils, They can just use Cygwin

**Comment by sylveon** [comment]
This is because git directly depends on those GNU utils. I don't think this will change anytime soon, since git for Windows attempts to keep the amount of changes to a minimum to keep up to date with upstream git (which is a Linux-specific project).

This seems more like a failure of winget for not exposing configuration options for git's installer and having a bad default.

**Reply by just1a-person** [reply]
This still doesn't explain the need of MinTTY and Git Bash

**Reply by sylveon** [reply]
There are options in git's installer to avoid installing both, this is also a winget failure for not exposing the options.

**Reply by just1a-person** [reply]
With a tool like winget I expect to be able to just `winget install git` and not have to worry about the installer options. I think this is a problem of the defaults

**Reply by sylveon** [reply]
You can go file feedback about the defaults into their git repo: https://github.com/git-for-windows/git

git for Windows isn't a Microsoft project so there isn't much they can do here except maybe changing winget to disable those options when installing git (which would also make some people relying on those tools grumpy)

**Reply by just1a-person** [reply]
Sorry for the late reply

I will wait for others to comment on this discussion. I want to know if others feel the same. Then I will think of submitting a new issue at [git-for-windows/git](https://github.com/git-for-windows/git)

**Reply by zooba** [reply]
Personally, I always disable those options when I install.

If the Git installer supports command line options for them, there could be a second manifest for Winget to install a "minimally impactful" copy of Git. That would go in the Winget repo, though Git may need to implement setup options.
